src/components/Auth/Signup.js

```diff
@@ -0,0 +1,129 @@
+import axios from "axios";
+import React from "react";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useAuthContex } from "../../contex/auth-contex";
+import "./auth.css";
+
+const Signup = () => {
+  const navigateHome = useNavigate();
+  const { authDispatch } = useAuthContex();
+  const [userDetail, setUserDetail] = useState({
+    name: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+  let name, value;
+  const setUserDetailHandler = (e) => {
+    name = e.target.name;
+    value = e.target.value;
+    setUserDetail({ ...userDetail, [name]: value });
+  };
+
+  const submitUserDetail = async (e) => {
+    e.preventDefault();
+    if (userDetail.password === userDetail.confirmPassword) {
+      try {
+        const response = await axios.post(`/api/auth/signup`, userDetail);
+        authDispatch({
+          type: "SIGN_UP",
+          payload: {
+            token: response.data.encodedToken,
+            userData: userDetail,
+          },
+        });
+        localStorage.setItem("token", response.data.encodedToken);
+        localStorage.setItem("user", JSON.stringify(response.data));
```

Comment:
No need to store user data in localstorage if you are storing it in context

Reply:
i will remove these things later, as i m doing this for testing as of now